MPF.md

```diff
@@ -239,6 +239,46 @@ manage High Availability of Enterprise Hubs is enabled.
 
 **Note:** This class is **not** defined by default.
 
+### Disable plain http for CFEngine Enterprise Mission Portal
+
+By default Mission Portal listens for HTTP requests on port 80, redirecting to HTTPS on port 443. To prevent the web server from listening on port 80 at all define `default:cfe_cfengine_enterprise_disable_plain_http`.
+
+**For example:**
+
+```json
+{
+  "classes": {
+    "default:cfe_enterprise_disable_plain_http": {
+      "class_expressions": [ "am_policy_hub|policy_server::" ]
+    }
+  }
+}
+```
+
+**History:**
+
+- Added in CFEngine 3.23.0
+
+### Disable plain http redirect to https for CFEngine Enterprise Mission Portal
```

Comment:
This second section doesn't make sense to me: "Disable plain http redirect" and then `default:cfe_cfengine_enterprise_enable_plain_http` but that says enable_plain_http which would seem to enable listening on port 80 and I suppose by some configuration somewhere else, redirect. I don't see how this augment class impacts the webserver configuration in the way described.

Reply:
after looking, I see it in the template:

```    {{^classes.cfe_enterprise_enable_plain_http}}
    # Force https with redirection
    RewriteCond %{HTTPS} off
    RewriteRule (.*) https://%{HTTP_HOST}%{REQUEST_URI} [R=301,L]
    {{/classes.cfe_enterprise_enable_plain_http}}
```

The language and class names are a bit confusing here, especially since the template checks for a NOT condition has to do with HTTP to HTTPS redirection and not enabling plain HTTP really. ;)

Could we rename `cfe_cfengine_enterprise_enable_plain_http` to `cfe_cfengine_enterprise_disable_http_redirect_to_https` or similar?

Reply:
Yes, the names are a bit weird.

`cfe_cfengine_enterprise_enable_plain_http` has been there since 3.6.0 but it just was not listed in this documentation.  When that class is defined the http to https redirect is disabled, "enabling the use of plain http".

if you think it's important I can change from `cfe_cfengine_enterprise_enable_plain_http` to `cfe_cfengine_enterprise_disable_http_redirect_to_https`. It would need an additional promise, probably in def.cf to set `cfe_cfengine_enterprise_enable_plain_http` when `cfe_cfengine_enterprise_disable_http_redirect_to_https` is defined for backward compatibility.

Reply:
Adjustments made, please re-review.